File: backend/app/schemas/instagram_account.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

from app.domain.enums import InstagramAccountStatus


class InstagramAccountCreate(BaseModel):
    ig_user_id: str = Field(min_length=1, max_length=64)
    username: str = Field(min_length=1, max_length=255)
    access_token: str = Field(min_length=1)
    page_id: str | None = Field(default=None, max_length=64)
    token_expires_at: datetime | None = None
    webhook_enabled: bool = False


class InstagramAccountRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    shop_id: UUID
    ig_user_id: str
    page_id: str | None
    username: str
    token_expires_at: datetime | None
    webhook_enabled: bool
    status: InstagramAccountStatus
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/instagramAccount.ts
export type InstagramAccountStatus = 'connected' | 'disconnected' | 'expired';

export interface InstagramAccount {
  id: string;
  shop_id: string;
  ig_user_id: string;
  page_id: string | null;
  username: string;
  token_expires_at: string | null;
  webhook_enabled: boolean;
  status: InstagramAccountStatus;
  created_at: string;
  updated_at: string;
}

export interface InstagramAccountCreate {
  ig_user_id: string;
  username: string;
  access_token: string;
  page_id?: string;
  webhook_enabled?: boolean;
}
